### DISCUSSION TITLE
custom space unit won't work with overrides

### DISCUSSION THREAD

**Discussion by kevinxh** [chakra-ui/chakra-ui#3556] [Q&A] [answered]
Hi folks,

I'm having an issue with the style overriding system and I wonder if this is a bug.

Here is the story:

For my project, the minimum accessible button size is 44px (2.75rem), in order to achieve that, we need to override the Button component `md` size (from 40px to 44px.)

Here is what we did, and things don't work as expected:
1. we created a new spacing unit `11`, and we use the `extendTheme` api to extend the spacings. This step works, we verified it using `console.log(useTheme())` and we can see there is a spacing unit `11` that has the correct value
<img width="167" alt="Screen Shot 2021-03-11 at 12 47 01 PM" src="https://user-images.githubusercontent.com/10948652/110852987-e929b880-8267-11eb-91ce-950756662993.png">

2. we then create style overrides for Button:
```
const button = {
    // styles for different sizes ("sm", "md", "lg")
    sizes: {
        md: {
            h: 11,
            minW: 11,
            fontSize: 'md',
            px: 4
        }
    }
}
```
notice we want to use the new unit 11. we also verified it the overriding works:
<img width="469" alt="Screen Shot 2021-03-11 at 12 48 12 PM" src="https://user-images.githubusercontent.com/10948652/110853130-14140c80-8268-11eb-9989-6480c03c9ece.png">

3. however, if we inspect the button on the page, we notice the actual height is `11px` instead of `2.75rem`
<img width="220" alt="Screen Shot 2021-03-11 at 12 49 12 PM" src="https://user-images.githubusercontent.com/10948652/110853203-327a0800-8268-11eb-9e96-2d3f0b8742db.png">

This makes us think there is a bug with the overriding system. Please advise.

**Comment by malashevskyi** [comment, marked as answer]
Maybe I am wrong, but I think that spaces don`t relate to height and width, it is rather 'sizes' not 'spaces'. I tried with padding and margin and it is working.
![](https://i.imgur.com/hObsFEr.png)
<br>
If you want to use value for 'height' and 'width' you need to use vars from sizes: 
![](https://i.imgur.com/lauxmvw.png)
<br>
and also you can add your units:
![](https://i.imgur.com/ebJNq1u.png)

**Comment by kevinxh** [comment]
@malashevskyi You are probably right! thanks for catching that!

I guess a related point with the design of the theming system, https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/foundations/sizes.ts if you look at the default sizes, it import the spacing from the defaults, which means, if you override the spacings but not the sizes, the sizes will still include the default spacing.

<img width="341" alt="Screen Shot 2021-03-12 at 10 53 57 AM" src="https://user-images.githubusercontent.com/10948652/110985507-5fd6bc80-8321-11eb-921a-1fddae5fc18b.png">

The expected behavior should be that if you override the spacing units, the sizes should also change.
